test/Microsoft.Extensions.DependencyModel.Tests/DependencyContextJsonWriterTests.cs

```diff
@@ -0,0 +1,356 @@
+using FluentAssertions;
+
+namespace Microsoft.Extensions.DependencyModel.Tests
+{
+    public class DependencyContextJsonWriterTests
+    {
+        public JObject Save(DependencyContext dependencyContext)
+        {
+            using (var memoryStream = new MemoryStream())
+            {
+                new DependencyContextWriter().Write(dependencyContext, memoryStream);
+                using (var readStream = new MemoryStream(memoryStream.ToArray()))
+                {
+                    using (var textReader = new StreamReader(readStream))
+                    {
+                        using (var reader = new JsonTextReader(textReader))
+                        {
+                            return JObject.Load(reader);
+                        }
+                    }
+                }
+            }
+        }
+
+        public DependencyContext Create(
+            string target = null,
+            string runtime = null,
+            bool? isPortable = null,
+            CompilationOptions compilationOptions = null,
+            CompilationLibrary[] compileLibraries = null,
+            RuntimeLibrary[] runtimeLibraries = null,
+            IReadOnlyList<KeyValuePair<string, string[]>> runtimeGraph = null)
+        {
+            return new DependencyContext(
+                            target ?? string.Empty,
+                            runtime ?? string.Empty,
+                            isPortable ?? false,
+                            compilationOptions ?? CompilationOptions.Default,
+                            compileLibraries ?? new CompilationLibrary[0],
+                            runtimeLibraries ?? new RuntimeLibrary[0],
+                            runtimeGraph ?? new KeyValuePair<string, string[]>[0]
+                            );
+        }
+
+        [Fact]
+        public void SavesRuntimeGraph()
+        {
+            var result = Save(Create(
+                            "Target",
+                            "Target/runtime",
+                            runtimeGraph: new[]
+                            {
+                                new KeyValuePair<string, string[]>("win7-x64", new [] { "win6", "win5"}),
+                                new KeyValuePair<string, string[]>("win8-x64", new [] { "win7-x64"}),
+                            }));
+
+            var runtimes = result.Should().HaveProperty("runtimes")
+                .Subject.Should().BeOfType<JObject>().Subject;
+
+            var rids = runtimes.Should().HaveProperty("Target")
+                .Subject.Should().BeOfType<JObject>().Subject;
+
+            rids.Should().HaveProperty("win7-x64")
+                .Subject.Should().BeOfType<JArray>()
+                .Which.Values<string>().ShouldBeEquivalentTo(new[] { "win6", "win5" });
+
+            rids.Should().HaveProperty("win8-x64")
+                .Subject.Should().BeOfType<JArray>()
+                .Which.Values<string>().ShouldBeEquivalentTo(new[] { "win7-x64" });
+        }
```

Comment:
I think I'd like one or two tests that assert the actual JSON emitted. Since it's a little brittle, I'm fine with just one, but I'd like to see at least one test that asserts the actual outputted JSON string, for end-to-end verification.